### PR TITLE
Testing /ghissue #21

### PR DESCRIPTION
**Pull request by Sarthak5598**
#21

### CODE REVIEW

File: src/sammich/plugins/create_issue.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+class CreateIssuePlugin(MachineBasePlugin):
+    @command("/ghissue")
+    async def createissue(self, command):
+        channel_id = command._cmd_payload["channel_id"]
+        title = command.text.strip()
+
+        # Replace with your repository details
+        owner = "OWASP-BLT"
+        repo = "BLT-Sammich"
```
Comment (arkid15r) [inline src/sammich/plugins/create_issue.py:19]: These can be the class level attributes.

Or just inline them as you use them just once

It'd also make sense to make it configurable via settings if we want it to be easily used by others.

File: src/sammich/plugins/create_issue.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        # Issue data: only title
+        issue_data = {
+            "title": title,
+            "body": "",
+            "assignee": None,
+            "milestone": None,
+        }
+
+        try:
+            response = requests.post(url, headers=headers, json=issue_data)
```
Comment (arkid15r) [inline src/sammich/plugins/create_issue.py:40]: Consider (re-)using the [GitHub API client](https://github.com/PyGithub/PyGithub)'s code.

File: src/sammich/plugins/create_issue.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+github_token = secrets["github_token"]
+
+
+class CreateIssuePlugin(MachineBasePlugin):
+    @command("/ghissue")
+    async def createissue(self, command):
+        channel_id = command._cmd_payload["channel_id"]
+        title = command.text.strip()
+
+        g = Github(github_token)
```
Comment (arkid15r) [inline src/sammich/plugins/create_issue.py:21]:
```suggestion
        github = Github(github_token)
```

File: src/sammich/plugins/create_issue.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+        try:
+            repo = g.get_repo(f"{settings.owner}/{settings.repo}")
+            issue = repo.create_issue(title=title)
+            issue_url = issue.html_url
+            await self.say(channel_id, f"Issue created successfully: {issue_url}")
```
Comment (arkid15r) [inline src/sammich/plugins/create_issue.py:27]:
```suggestion
            await self.say(channel_id, f"Issue created successfully: {issue.html_url}")
```

File: src/sammich/plugins/create_issue.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        except Exception as e:
+            logging.error(f"Failed to create issue: {str(e)}")
+            await self.say(channel_id, f"Failed to create issue: {str(e)}")
```
Comment (arkid15r) [inline src/sammich/plugins/create_issue.py:30]: I guess just a failure message with no details would be enough.

File: src/settings.py [hunk 6]
```diff
@@ -0,0 +1,2 @@
+repo = "BLT-Sammich"
+owner = "OWASP-BLT"
```
Comment (arkid15r) [inline src/settings.py:2]:
```suggestion
GITHUB_REPOSITORY_OWNER = "OWASP-BLT"
GITHUB_REPOSITORY_NAME = "BLT-Sammich"
```

File: src/sammich/plugins/create_issue.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+from dotenv import dotenv_values
+from github import Github
+from machine.plugins.base import MachineBasePlugin
+from machine.plugins.decorators import command
+
+import settings
+
+# Load secrets
+secrets = dotenv_values(".secrets")
+github_token = secrets["github_token"]
```
Comment (arkid15r) [inline src/sammich/plugins/create_issue.py:12]: Why do you need it here?

**Review by arkid15r — CHANGES_REQUESTED** [review]

File: src/sammich/plugins/create_issue.py (hunk 7 above)
Comment (Sarthak5598) [reply src/sammich/plugins/create_issue.py:12]: where should I add it inside the function!!

Comment (arkid15r) [reply src/sammich/plugins/create_issue.py:12]: To a right place depending on the scope needed by your code. Check where you use it first.

Comment (Sarthak5598) [reply src/sammich/plugins/create_issue.py:12]: how about now !?

File: src/sammich/plugins/create_issue.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+secrets = dotenv_values(".secrets")
+
+
+class CreateIssuePlugin(MachineBasePlugin):
+    @command("/ghissue")
+    async def createissue(self, command):
+        channel_id = command._cmd_payload["channel_id"]
+        title = command.text.strip()
+        github_token = secrets["github_token"]
+        github = Github(github_token)
```
Comment (arkid15r) [inline src/sammich/plugins/create_issue.py:19]:
```suggestion
        github = Github(secrets["github_token"])
```

**Comment by Sarthak5598** [comment]
Thanks for all the help @arkid15r !, I thinks its ready to merge now !

**Comment by arkid15r** [comment]
> Thanks for all the help @arkid15r !, I thinks its ready to merge now !

It seems tests are missing. Are you going to add them? Maybe in another PR?

**Comment by Sarthak5598** [comment]
I will tests for both this and reminder once both are done !

**Review by arkid15r — APPROVED** [review]

**Pull request merged**